logstash/tests/logstash_test.py

```diff
@@ -307,6 +307,111 @@ def test_adding_a_secret_mount_with_subpath():
     }
 
 
+def test_adding_a_secret():
+    config = """
+secretCreate:
+  - name: "env"
+    value:
+      ELASTICSEARCH_PASSWORD: "LS1CRUdJTiBgUFJJVkFURSB"
+"""
+    r = helm_template(config)
+    secret_name = name + "-env"
+    s = r["secret"][secret_name]
+    assert s["metadata"]["labels"]["app"] == name
+    assert len(r["secret"]) == 1
+    assert len(s["data"]) == 1
+    assert s["data"] == {"ELASTICSEARCH_PASSWORD": "TFMxQ1JVZEpUaUJnVUZKSlZrRlVSU0I="}
+
+
+def test_adding_secret_from_file():
+    config = """
+secretCreate:
+  - name: "tls"
+    value:
+      cert.key.filepath: "secrets/private.key"
+"""
+    content = """
+-----BEGIN RSA PRIVATE KEY-----
+MIIEpAIBAAKCAQEApCt3ychnqZHsS
+DylPFZn55xDaDcWco1oNFdBGzFjw+
+zkuMFMOv7ab+yOFwHeEeAAEkEgy1u
+Da1vIscBs1K0kbEFRSqySLuNHWiJp
+wK2cI/gJc+S9Qd9Qsn0XGjmjQ6P2p
+ot2hvCOtnei998OmDSYORKBq2jiv/
+-----END RSA PRIVATE KEY-----
+"""
+    work_dir = os.path.join(os.path.abspath(os.getcwd()), "secrets")
+    filename = os.path.join(work_dir, "private.key")
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
+    with open(filename, "w") as f:
+        f.write(content)
+
+    with open(filename, "r") as f:
+        data = f.read()
+        assert data == content
+
+    r = helm_template(config)
+    secret_name = name + "-tls"
+    s = r["secret"][secret_name]
+    assert s["metadata"]["labels"]["app"] == name
+    assert len(r["secret"]) == 1
+    assert len(s["data"]) == 1
+    assert s["data"] == {
+        "cert.key": "Ci0tLS0tQkVHSU4gUlNBIFBSSVZBVEUgS0VZLS0tLS0KTUlJRXBBSUJBQUtDQVFFQXBDdDN5Y2hucVpIc1MKRHlsUEZabjU1eERhRGNXY28xb05GZEJHekZqdysKemt1TUZNT3Y3YWIreU9Gd0hlRWVBQUVrRWd5MXUKRGExdklzY0JzMUswa2JFRlJTcXlTTHVOSFdpSnAKd0syY0kvZ0pjK1M5UWQ5UXNuMFhHam1qUTZQMnAKb3QyaHZDT3RuZWk5OThPbURTWU9SS0JxMmppdi8KLS0tLS1FTkQgUlNBIFBSSVZBVEUgS0VZLS0tLS0K",
```

Comment:
Could we switch this value to a var that's generated from the `content`?

That way, if the `content` ever changes the test will still work.

Something like this should work:
```python
import base64
...
content = ...
content_b64 = base64.b64encode(content.encode('ascii'))
...
"cert.ket": content_b64
```

Reply:
Ok, good idea ! I'll do it.